servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TimeoutPublisher.java

```diff
@@ -282,13 +282,16 @@ private void offloadTimeout(Throwable cause) {
         private void processTimeout(Throwable cause) {
             final Subscription subscription = subscriptionUpdater.getAndSet(this, EMPTY_SUBSCRIPTION);
             // The timer is started before onSubscribe so the subscription may actually be null at this time.
-            if (subscription != null) {
-                subscription.cancel();
-                // onErrorFromTimeout will protect against concurrent access on the Subscriber.
-            } else {
-                target.onSubscribe(EMPTY_SUBSCRIPTION);
+            try {
+                if (subscription != null) {
+                    subscription.cancel();
+                    // onErrorFromTimeout will protect against concurrent access on the Subscriber.
+                } else {
+                    target.onSubscribe(EMPTY_SUBSCRIPTION);
+                }
+            } finally {
+                target.processOnError(cause);
```

Comment:
Why `finally` is required here? For cases when `onSubscribe/cancel` throws?

Reply:
Yes, to ensure that the original "true cause" is not lost.